src/build/env.py

```diff
@@ -187,6 +188,25 @@ def _create_isolated_env_virtualenv(path):  # type: (str) -> Tuple[str, str]
 # venv only exists on Python 3+
 if sys.version_info >= (3,):  # noqa: C901
 
+    once = functools.lru_cache(maxsize=None)
+
+    @once
```

Comment:
Why are we assigning it to a variable? Did I missing something?

Reply:
This was discussed as an addition to functools, where it would have been called `functools.once`. It didn't make it very far since `once = functools.lru_cache(maxsize=None)` is hard to improve on in terms of performance (it's internally implemented in C) and is only one line long (and would actually be `once = functools.cache` in Python 3.9+). Happy to use it directly instead though if you want to. This highlights what it does _slightly_ better, I think - it's only evaluated once. The `maxsize=None` could be confusing without the rename; it's not actually growing, it just is needed to remove the eviction code.

Reply:
Before I change this, do you want:

1. Current `once = ` name
2. Use `_once = ` name
3. Use directly, no variable
4. Use directly, no variable, but have a comment saying this is there to cache the value once on first call.

I like options 1-3 best personally; I don't really like 4 since options 1 & 2 do the same thing but use a named variable instead of a comment.

Reply:
I used Option 3 in the current PR.

Reply:
This is fine, though it was not obvious to me. I would just apply the decorator directly, but no need to change the commit just for that.

Reply:
That's what I did in the current commit, that was option 3 :)